dash/include/dash/matrix/internal/LocalMatrixRef-inl.h

```diff
@@ -445,28 +445,28 @@ ::sub(
 
 template<typename T, dim_t NumDim, dim_t CUR, class PatternT>
 inline LocalMatrixRef<T, NumDim, NumDim-1, PatternT>
-LocalMatrixRef<T, NumDim, CUR, PatternT>
+const LocalMatrixRef<T, NumDim, CUR, PatternT>
```

Comment:
We cannot return references, as the functions return temporaries. I made them constexpr whenever possible in the next commit.